writer.go

```diff
@@ -859,10 +860,13 @@ func (w *writer) writeRowGroup(rowGroupSchema *Schema, rowGroupSortingColumns []
 			c.offsetIndex.PageLocations[j].Offset += dataPageOffset
 		}
 
-		for _, page := range c.pages {
-			if _, err := io.Copy(&w.writer, page); err != nil {
-				return 0, fmt.Errorf("writing buffered pages of row group column %d: %w", i, err)
-			}
+		if offset, err := c.pageBuffer.Seek(0, io.SeekStart); err != nil {
+			return 0, err
+		} else if offset != 0 {
+			return 0, fmt.Errorf("resetting parquet page buffer to the start expected offset zero but got %d", offset)
+		}
+		if _, err := io.Copy(&w.writer, c.pageBuffer); err != nil {
```

Comment:
We could use `io.CopyBuffer` and reuse the buffer across columns instead of using a new one for each column.

Reply:
Not a bad idea, though io.Copy (and CopyBuffer) both use WriterTo or ReaderFrom if the src/dst implement them. I don't know how often this actually winds up using the buffer during the copy anyway, and we'd want to avoid allocating a buffer at all if it won't wind up being used.

Reply:
Make sense.